Makefile

```diff
@@ -171,7 +173,7 @@ create-cluster: ## Create a Kubernetes cluster on AWS using examples
 	--bootstrap-type kind \
 	-m ./cmd/clusterctl/examples/aws/out/machines.yaml \
 	-c ./cmd/clusterctl/examples/aws/out/cluster.yaml \
-	-p ./cmd/clusterctl/examples/aws/out/provider-components.yaml \
+	-p ./cmd/clusterctl/examples/aws/out/provider-components-dev.yaml \
```

Comment:
Considering this is using the `-dev` resources here, should this target be renamed to `create-cluster-dev` instead?

Reply:
since we've now brought make out of the mix for users, I think the implication is that it will always be a dev target